.github/workflows/ci.yml

```diff
@@ -2,145 +2,186 @@ name: Continuous Integration
 
 on: [push, pull_request, workflow_dispatch]
 
+env:
+  TEST_ALL_DOCKER_IMAGE: 'parsec-service-test-all'
+  # TEST_ALL_DOCKER_IMAGE: 'ghcr.io/parallaxsecond/parsec-service-test-all'
+
 jobs:
+  build-and-export-test-all-docker:
+    runs-on: ubuntu-latest
+    # If TEST_ALL_DOCKER_IMAGE is 'parsec-service-test-all' or any local image,
+    # the following condition must evaluate true to execute this job
+    # Else it must evaluate false to NOT execute this job
+    # Unfortunately, env.TEST_ALL_DOCKER_IMAGE cannot be used here as the `env` context is not recognize at this level.
+    if: ${{ true }} # env.TEST_ALL_DOCKER_IMAGE == 'parsec-service-test-all'
+    steps:
+    - uses: actions/checkout@v3
+    - name: Build the docker container
+      run: pushd e2e_tests/docker_image && docker build -t parsec-service-test-all -f parsec-service-test-all.Dockerfile . && popd
+    - name: Export the docker container
+      run: docker save parsec-service-test-all > /tmp/parsec-service-test-all.tar
+    - name: Upload artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: parsec-service-test-all
+        path: /tmp/parsec-service-test-all.tar
+
   all-providers:
     name: Various tests targeting a Parsec image with all providers included
     runs-on: ubuntu-latest
+    if: ${{ always() }}
+    needs: [build-and-export-test-all-docker]
```

Comment:
Does this mean we'll always build the image on the CI?

Reply:
it's the other way around
it will always run even if the `build-and-export-test-all-docker` is skipped